crates/ir/src/lir.rs

```diff
@@ -0,0 +1,70 @@
+use std::io::Write;
+
+use serde::{Deserialize, Serialize};
+use zip::{write::FileOptions, ZipWriter};
+
+use crate::MirNodeId;
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[repr(transparent)]
+pub struct LirNodeId(String);
+
+impl LirNodeId {
+    pub fn new(id: &str) -> Self {
+        Self(id.to_string())
+    }
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[repr(transparent)]
+pub struct LirStreamId(usize);
+
+impl LirStreamId {
+    pub fn new(id: usize) -> Self {
+        Self(id)
+    }
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct LirNode {
+    pub id: LirNodeId,
+
+    /// Node type, e.g., map, join, etc.
+    pub operation: String,
+
+    /// The list of Mir node ids that this LIR node implementa completely or partially.
+    #[serde(default)]
+    pub implements: Vec<MirNodeId>,
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct LirEdge {
+    /// Stream id if this edge is a stream edge. None if this is a dependency edge.
+    /// Dependency edges connect operators that implement a single logical function,
+    /// e.g., exchange sender and receiver or the input and output halves of Z1.
+    pub stream_id: Option<LirStreamId>,
```

Comment:
I think you will want to represent the output number too

Reply:
I don't have them now; can probably refactor things to support numbers if needed.